extensions/standard-processors/controllers/UnorderedMapPersistableKeyValueStoreService.h

```diff
@@ -59,20 +61,30 @@ class UnorderedMapPersistableKeyValueStoreService : public AbstractAutoPersistin
 
   bool update(const std::string& key, const std::function<bool(bool /*exists*/, std::string& /*value*/)>& update_func) override;
 
-  bool persist() override;
+  bool persist() override {
+    persistNonVirtual();
+  }
 
  protected:
+  using AbstractAutoPersistingKeyValueStoreService::getImpl;
+  using AbstractAutoPersistingKeyValueStoreService::setImpl;
+  using AbstractAutoPersistingKeyValueStoreService::persistImpl;
+  using AbstractAutoPersistingKeyValueStoreService::removeImpl;
+
+
   static constexpr const char* FORMAT_VERSION_KEY = "__UnorderedMapPersistableKeyValueStoreService_FormatVersion";
   static constexpr int FORMAT_VERSION = 1;
 
   std::string file_;
 
   bool load();
 
-  std::string escape(const std::string& str);
   bool parseLine(const std::string& line, std::string& key, std::string& value);
 
  private:
+  void initializeNonVirtual();
+  bool persistNonVirtual();
```

Comment:
Why are these necessary? If subclasses need to call the base class implementation, they can always qualify the call with the base class name to avoid dynamic dispatching,

Reply:
Because initialize() and persist() were called in constructor/destructor and they are virtual methods.